### ISSUE TITLE
no-constant-condition could check yield in generators

### ISSUE THREAD

**Issue by samsch** [eslint/eslint#7321]
**Tell us about your environment**
- **ESLint Version: 3.5.0 & 3.7.1**
- **Node Version: 6.6.0**
- **npm Version: 3.10.9**

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**

```
env:
  browser: true
  es6: true
  node: true
extends: 'eslint:recommended'
rules:
  indent:
    - error
    - 4
  linebreak-style:
    - error
    - unix
  quotes:
    - error
    - single
  semi:
    - error
    - always
```

**What did you do? Please include the actual source code causing the issue.**

``` js
const g = function *() {

    const a = 0;

    while (true) {
        yield a++;
    }
};
```

**What actually happened? Please include the actual, raw output from ESLint.**
Currently: `5:5  error  Unexpected constant condition  no-constant-condition`

Would it be possible to have an option for no-constant-condition allowing `while(true)` when the block contains a `yield`? I suppose it might be reasonable to have that option for `break` as well.

Or am I getting no-constant-condition too mixed up with no-unreachable?

**Comment by platinumazure** [comment]
For what it's worth, if you use `for (;;)` or similar, that will not warn.

**Comment by not-an-aardvark** [comment]
I see why one would want to do this, but I feel like it contradicts the point of the rule. The point of the rule is to make sure the while loop has no constant conditions, and the while loop in your function has a constant condition.

The only difference from other `while (true)` loops is that yours doesn't block the event loop, but I don't think that should be a determining factor in whether this rule warns.

**Comment by kaicataldo** [comment]
Closing this as discussion has stalled and it doesn't appear there is enough interest, either from the team or from the community, to continue the discussion at this time. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. Please feel free to open a new issue with a proposal to continue this discussion if this is something you would like to continue advocating for.
